src/lib/Persistence/Legacy/Content/Type/Gateway/CriterionHandler/ContentTypeGroupName.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Core\Persistence\Legacy\Content\Type\Gateway\CriterionHandler;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\ParameterType;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Ibexa\Contracts\Core\Persistence\Content\Type\CriterionHandlerInterface;
+use Ibexa\Contracts\Core\Repository\Values\ContentType\Query\Criterion\ContentTypeGroupName as ContentTypeGroupNameCriterion;
+use Ibexa\Contracts\Core\Repository\Values\ContentType\Query\CriterionInterface;
+use Ibexa\Core\Persistence\Legacy\Content\Type\Gateway;
+use Ibexa\Core\Persistence\Legacy\Content\Type\Gateway\CriterionVisitor\CriterionVisitor;
+
+/**
+ * @implements \Ibexa\Contracts\Core\Persistence\Content\Type\CriterionHandlerInterface<\Ibexa\Contracts\Core\Repository\Values\ContentType\Query\Criterion\ContentTypeGroupName>
+ */
+final class ContentTypeGroupName implements CriterionHandlerInterface
+{
+    public function supports(CriterionInterface $criterion): bool
+    {
+        return $criterion instanceof ContentTypeGroupNameCriterion;
+    }
+
+    /**
+     * @param \Ibexa\Contracts\Core\Repository\Values\ContentType\Query\Criterion\ContentTypeGroupName $criterion
+     */
+    public function apply(
+        CriterionVisitor $criterionVisitor,
+        QueryBuilder $qb,
+        CriterionInterface $criterion
+    ): string {
+        $subQuery = $qb->getConnection()->createQueryBuilder();
+        $whereClause = is_array($criterion->getValue())
+            ? $subQuery->expr()->in(
+                'LOWER(ctg.name)',
+                $qb->createNamedParameter(array_map('strtolower', $criterion->getValue()), Connection::PARAM_STR_ARRAY)
+            ) : $subQuery->expr()->eq(
+                'LOWER(ctg.name)',
+                $qb->createNamedParameter(strtolower($criterion->getValue()), ParameterType::STRING)
+            );
```

Comment:
How about this approach instead? For databases, it does not matter if it's equality or `IN`, they can deal with it accordingly without performance penalties.
```suggestion
        $subQuery = $qb->getConnection()->createQueryBuilder();
        $value = $criterion->getValue();
        if (!is_array($value) {
            $value = [$value];
        }
        $whereClause = $subQuery->expr()->in(
            'LOWER(ctg.name)',
            $qb->createNamedParameter(array_map('strtolower', $criterion->getValue()), Connection::PARAM_STR_ARRAY)
        );
```